### DISCUSSION TITLE
4.1 Set-up a local development environment

### DISCUSSION THREAD

**Discussion by mdminhaj88** [opencrvs/opencrvs-core#6634] [Technical Q&A]
### OpenCRVS Core version

1.4.0

### OpenCRVS Countryconfig version

1.4.0

### Your operating system and version

Ubuntu 22.04

### Your node version

v16.20.0

### Which of these describes your project best?

- [X] Testing OpenCRVS for business development, educational purposes or for personal interest.
- [ ] A research project to evaluate, configure and test OpenCRVS as a possible solution for a government, an active brief or proposal.
- [ ] Actively installing and configuring OpenCRVS Core for production use in a country.
- [ ] Customising OpenCRVS Core code in production in a country

**Comment by mdminhaj88** [comment]
hi @rikukissa,

Hope you are doing good

Was trying to setup a local dev environment on my laptop by following the instructions but as i run the script 'bash setup.sh' it is throwing an error that operating system not supported.

Whereas i am using a WSL ubuntu on my windows laptop

attached is the ss of the same.

Thank you

![ubuntu](https://github.com/opencrvs/opencrvs-core/assets/142885612/b718f8ef-4151-4fcb-aae2-fd30e2ae05bc)

**Reply by rikukissa** [reply]
Hey @mdminhaj88! Does it work if you try it in the `develop` branch? This is something we fixed very recently
https://github.com/opencrvs/opencrvs-core/blob/develop/setup.sh#L122

**Reply by mdminhaj88** [reply]
It worked when i did try with the develop branch @rikukissa ! 

But later during the setup its failing by throwing an error, im not sure why its happening.

please find the ss attached.

Thanks

![error-local-dev](https://github.com/opencrvs/opencrvs-core/assets/142885612/6557db4e-6cdf-4b3a-a41d-0d76495b1037)

**Reply by mdminhaj88** [reply]
and when i accessed the url, its struck at loading as can be seen in the attached ss

![stuck at loading](https://github.com/opencrvs/opencrvs-core/assets/142885612/8340dd99-2a21-4553-ae32-1433abd246db)

**Reply by rikukissa** [reply]
Seems like a random error..  Can you try to focus the right bottom terminal window and run 

```
bash development-environment/setup-countryconfig.sh $(pwd)
```
